contracts/access/AccessControl.sol

```diff
@@ -164,12 +164,17 @@ abstract contract AccessControl is Context {
      * event. Note that unlike {grantRole}, this function doesn't perform any
      * checks on the calling account.
      *
-     * Requirements:
-     *
-     * - this function can only be called from a constructor.
+     * [WARNING]
+     * ====
+     * This function should only be called from the constructor when setting
+     * up the initial roles for the system.
+     *
+     * It being callable by other means goes against the spirit of
+     * {AccessControl}, as there would be ways to grant roles not tracked by
+     * {getRoleMemberCount} and {getRoleMember}.
```

Comment:
```suggestion
     * Using this function in any other way is effectively circumventing the admin
     * system imposed by {AccessControl}.
```

Putting it this way, I'm getting the feeling that we should rename the function to something like `_unsafeGrantRole`.

Reply:
An alternative idea I just had is to allow `_setupRole` only when the member count for the role is zero. In that case though we may need a function to add multiple accounts to the role.

Reply:
I think your wording is better, yes.

The issue with checking for zero members is that may naturally happen during the lifetime of the contract as roles are revoked and renounced.